perception/sandbox/polygrasp/src/polygrasp/grasp_rpc.py

```diff
@@ -0,0 +1,178 @@
+import logging
+import numpy as np
+import open3d as o3d
+from matplotlib import pyplot as plt
+
+import a0
+
+import graspnetAPI
+from polygrasp import serdes
+from polygrasp.serdes import polygrasp_msgs
+
+
+log = logging.getLogger(__name__)
+topic_key = "grasp_server"
+grasp_topic_key = f"{topic_key}/grasp"
+collision_topic_key = f"{topic_key}/collision"
+
+
+def save_img(img, name):
+    f = plt.figure()
+    plt.imshow(img)
+    f.savefig(f"{name}.png")
+    plt.close(f)
+
+
+class GraspServer:
+    def _get_grasps(self, pcd: o3d.geometry.PointCloud) -> np.ndarray:
+        raise NotImplementedError
+
+    def _get_collisions(
+        self, grasps: graspnetAPI.GraspGroup, scene_pcd: o3d.geometry.PointCloud
+    ) -> graspnetAPI.GraspGroup:
+        raise NotImplementedError
+
+    def start(self):
+        log.info(f"Starting grasp server...")
+
+        def grasp_onrequest(req):
+            log.info(f"Got request; computing grasp group...")
+
+            payload = req.pkt.payload
+            pcd = serdes.capnp_to_pcd(payload)
+            grasp_group = self._get_grasps(pcd)
+
+            log.info(f"Done. Replying with serialized grasp group...")
+            req.reply(serdes.grasp_group_to_capnp(grasp_group).to_bytes())
+
+        grasp_server = a0.RpcServer(grasp_topic_key, grasp_onrequest, None)
+
+        def collision_onrequest(req):
+            log.info(f"Got request; computing collisions...")
+
+            payload = req.pkt.payload
+            msg = polygrasp_msgs.CollisionRequest.from_bytes(payload)
+            grasp_group = serdes.bytes_to_grasp_group(msg.grasps)
+            scene_pcd = serdes.capnp_to_pcd(msg.pcd)
+
+            filtered_grasp_group = self._get_collisions(grasp_group, scene_pcd)
+            log.info(f"Done. Replying with serialized filtered grasps...")
+            req.reply(serdes.grasp_group_to_bytes(filtered_grasp_group))
+
+        collision_server = a0.RpcServer(collision_topic_key, collision_onrequest, None)
```

Comment:
If `grasp_server` and `collision_server` outlived the function, you can have the function return.
Maybe by just adding `self.`